casper/src/main/scala/coop/rchain/casper/engine/Running.scala

```diff
@@ -1,68 +1,38 @@
-import coop.rchain.comm.rp.ProtocolHelper.protocol
-import coop.rchain.comm.transport.{Blob, TransportLayer}
+import coop.rchain.comm.transport.TransportLayer
 import coop.rchain.comm.PeerNode
-import coop.rchain.metrics.Metrics
+import coop.rchain.metrics.{Metrics, MetricsSemaphore}
 import coop.rchain.models.BlockHash.BlockHash
-import coop.rchain.shared.{Cell, Log, Time}
-import coop.rchain.catscontrib.Catscontrib.ToBooleanF
+import coop.rchain.shared.{Log, Time}
 
+import scala.collection.concurrent.TrieMap
 import scala.concurrent.duration._
 
 object Running {
 
-  implicit private[this] val BlockRequesterMetricsSource =
-    Metrics.Source(CasperMetricsSource, "block-requester")
+  implicit val MetricsSource: Metrics.Source =
+    Metrics.Source(CasperMetricsSource, "running")
 
-  final case class Requested(
-      timestamp: Long,
-      peers: Set[PeerNode] = Set.empty,
-      received: Boolean = false,
-      waitingList: List[PeerNode] = List.empty
-  )
-
-  type RequestedBlocks[F[_]] = Cell[F, Map[BlockHash, Requested]]
-  object RequestedBlocks {
-    def apply[F[_]: RequestedBlocks]: RequestedBlocks[F] = implicitly[RequestedBlocks[F]]
-    def put[F[_]: RequestedBlocks](hash: BlockHash, requested: Requested) =
-      RequestedBlocks[F].modify(_ + (hash -> requested))
-    def remove[F[_]: RequestedBlocks](hash: BlockHash) = RequestedBlocks[F].modify(_ - hash)
-    def get[F[_]: RequestedBlocks](hash: BlockHash)    = RequestedBlocks[F].reads(_.get(hash))
-    def contains[F[_]: RequestedBlocks](hash: BlockHash) =
-      RequestedBlocks[F].reads(_.contains(hash))
-  }
-
-  private def sendToPeer[F[_]: Monad: RPConfAsk: TransportLayer](peer: PeerNode)(packet: Packet) =
-    RPConfAsk[F].ask
-      .map(conf => protocol(conf.local, conf.networkId).withPacket(packet))
-      .flatMap(msg => TransportLayer[F].send(peer, msg))
-
-  private def streamToPeer[F[_]: Monad: RPConfAsk: TransportLayer](peer: PeerNode)(packet: Packet) =
-    RPConfAsk[F].ask
-      .map(conf => Blob(conf.local, packet))
-      .flatMap(blob => TransportLayer[F].stream(peer, blob))
-
-  private def requestForBlock[F[_]: Monad: RPConfAsk: TransportLayer: RequestedBlocks](
-      peer: PeerNode,
-      hash: BlockHash
-  ) = sendToPeer(peer)(ToPacket(BlockRequestProto(hash)))
-
-  private def requestForNewBlock[F[_]: Monad: RPConfAsk: TransportLayer: RequestedBlocks: Log: Time](
-      peer: PeerNode,
-      blockHash: BlockHash
-  ) = requestForBlock(peer, blockHash) >> addNewEntry(blockHash, Some(peer))
+  trait CasperMessageStatus
+  final case object BlockIsInDag            extends CasperMessageStatus
+  final case object BlockIsInCasperBuffer   extends CasperMessageStatus
+  final case object BlockIsWaitingForCasper extends CasperMessageStatus
+  final case object DoNotIgnore             extends CasperMessageStatus
+  final case class IgnoreCasperMessageStatus(doIgnore: Boolean, status: CasperMessageStatus)
```

Comment:
It looks like `IgnoreCasperMessageStatus` is another message state but it not. It would be more clear with one empty line in between.